quickwit/quickwit-proto/Cargo.toml

```diff
@@ -25,6 +25,7 @@ prost-types = { workspace = true }
 serde = { workspace = true }
 serde_json = { workspace = true }
 sqlx = { workspace = true, optional = true }
+tantivy = { workspace = true }
```

Comment:
no.

Reply:
It's coming from the type conversion, which is used in indexing and search (schema fallback)
This is another variant of the quickwit-indexing->search dependency problem.

Since there is no common layer, I think there are 3 options, duplicate the conversion code, use tantivy to store list fields, add a common layer crate.

```rust
impl From<Type> for ListFieldType {
    fn from(typ: Type) -> Self {
        match typ {
            Type::Str => ListFieldType::Str,
            Type::U64 => ListFieldType::U64,
            Type::I64 => ListFieldType::I64,
            Type::F64 => ListFieldType::F64,
            Type::Bool => ListFieldType::Bool,
            Type::Date => ListFieldType::Date,
            Type::Facet => ListFieldType::Facet,
            Type::Bytes => ListFieldType::Bytes,
            Type::Json => ListFieldType::Json,
            Type::IpAddr => ListFieldType::IpAddr,
        }
    }
}
```

Reply:
on the consumption (search) side, do we even need to ever do conversion?
on the indexing side, we can have the conversion code.

Reply:
Yes, there's a fallback case in search to build a field list from `Schema`, if the list fields file does not exist

Reply:
I think we should remove the fallback, It's ok to say to the user that we returned nothing with splits built before adding this feature.

Reply:
I removed the fallback, it's a not great though to have a worse API to circumvent the dependency issue